solr/core/src/test/org/apache/solr/cloud/overseer/ZkStateReaderTest.java

```diff
@@ -29,35 +38,68 @@
 import org.apache.solr.cloud.ZkTestServer;
 import org.apache.solr.common.cloud.ClusterState;
 import org.apache.solr.common.cloud.DocCollection;
+import org.apache.solr.common.cloud.DocCollectionWatcher;
 import org.apache.solr.common.cloud.DocRouter;
 import org.apache.solr.common.cloud.SolrZkClient;
 import org.apache.solr.common.cloud.ZkStateReader;
+import org.apache.solr.common.util.ExecutorUtil;
+import org.apache.solr.common.util.SolrNamedThreadFactory;
 import org.apache.solr.common.util.TimeSource;
 import org.apache.solr.handler.admin.ConfigSetsHandler;
 import org.apache.solr.util.TimeOut;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class ZkStateReaderTest extends SolrTestCaseJ4 {
-
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
   private static final long TIMEOUT = 30;
 
-  public void testExternalCollectionWatchedNotWatched() throws Exception {
+  private static class TestFixture implements Closeable {
```

Comment:
This is an interesting approach. I meant to put things into a JUnit `@Before` and `@After` method, which is going to be more idiomatic, I think.

Reply:
Thanks! I tried using `@Before` and `@After` but it's a bit tricky as I wish to call `createTempDir` with different test name therefore we will need to pass some parameters to the @Before. I looked up and there's this https://stackoverflow.com/questions/26357751/passing-parameter-to-before-setup-in-junit-test but it seems a bit complicated for here.

Is it okay if we keep the current structure?

Reply:
You can use `getTestName()` to get the current test name as a string (inherited from LuceneTestCase) and pass that to `createTempDir` and it should all work out. Let me know if there are issues with that approach, and we can think about how to simplify if needed.

Reply:
Ah didn't know that! Thank you for the useful pointer!